src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CreditManager.cs

```diff
@@ -44,33 +44,27 @@ public ValueTask<int> RequestCreditAsync(int amount, CancellationToken cancellat
             {
                 if (_disposed)
                 {
-                    throw new ObjectDisposedException(nameof(CreditManager));
+                    throw CreateObjectDisposedException(forActiveWaiter: false);
                 }
 
                 if (_current > 0)
                 {
                     Debug.Assert(_waiters == null || _waiters.Count == 0, "Shouldn't have waiters when credit is available");
 
                     int granted = Math.Min(amount, _current);
+                    if (NetEventSource.IsEnabled) _owner.Trace($"{_name}. requested={amount}, current={_current}, granted={granted}");
                     _current -= granted;
                     return new ValueTask<int>(granted);
                 }
 
-                // Uses RunContinuationsAsynchronously internally.
-                var tcs = new TaskCompletionSourceWithCancellation<int>();
-
-                if (_waiters == null)
-                {
-                    _waiters = new Queue<Waiter>();
-                }
+                if (NetEventSource.IsEnabled) _owner.Trace($"{_name}. requested={amount}, no credit available.");
```

Comment:
Nit: seems like we could do the tracing in the caller, and avoid having to pass in owner/name to this object. But I don't have a strong objection to doing it this way.